### PR TITLE
feat: implement basic project structure

### PR DESCRIPTION
**Pull request by craigzour**
# Summary | Résumé

**CONSIDER IT AS A PROPOSAL**

- Implements basic project structure

### CODE REVIEW

File: src/middleware/globalErrorHandler/middleware.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Request, Response, NextFunction } from "express";
+
+function main(error: Error, _request: Request, response: Response, _next: NextFunction) {
+  console.log(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```
Comment (patheard) [inline src/middleware/globalErrorHandler/middleware.ts:4]: Could we do this like so instead?
```suggestion
  console.error(JSON.stringify(error, Object.getOwnPropertyNames(error)));
```

Comment (craigzour) [reply src/middleware/globalErrorHandler/middleware.ts:4]: Absolutely!

File: src/middleware/authentication/middleware.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { NextFunction, Request, Response } from "express";
+
+async function main(request: Request, response: Response, next: NextFunction) {
```
Comment (wmoussa-gc) [inline src/middleware/authentication/middleware.ts:3]: This is where we should verify the JWT key, correct?

Comment (craigzour) [reply src/middleware/authentication/middleware.ts:3]: Correct!

**Comment by craigzour** [comment]
@patheard @wmoussa-gc I have been testing https://biomejs.dev/ this afternoon to replace both eslint and prettier. Looking really good so far. Will make a separate PR for it. I will follow all their established rules to begin with and we will decide as a team what we want to disable in the future.

File: test/routes/forms/submission/router.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import express from "express";
+import router from "../../../../src/routes/forms/router";
+
+vi.mock("../../../../src/middleware/authentication/middleware", () => ({
+  default: (_req, _res, next) => next(),
+}));
+
+vi.mock("../../../../src/middleware/rateLimiter/middleware", () => ({
+  default: (_req, _res, next) => next(),
+}));
```
Comment (patheard) [inline test/routes/forms/submission/router.test.ts:12]: This is my quick/dirty solution to preventing middleware from blocking the router unit tests.  I'm sure there's a better way.

**Review by patheard — APPROVED** [review]

**Pull request merged**